retrace/src/proguard/retrace/ReTrace.java

```diff
@@ -226,6 +226,9 @@ private String handle(FrameInfo obfuscatedFrame, FrameRemapper mapper, FramePatt
                     }
 
                     result.append(trimmedLine);
+                    if (retracedFrames.hasNext()) {
+                        result.append("\n");
+                    }
```

Comment:
Based on the change in #200: it looks like this should be like the following to revert to the previous behaviour:

```suggestion
                    result.append(trimmedLine);
                    result.append(System.lineSeparator());
```

Reply:
Line 245 also had a similar change, which no longer adds a newline.

Reply:
I like the use of `System.lineSeparator()` instead of `"\n"` but if I accept your suggesion, we would end up with  a newline separating each batch of frames. For example:

```
09-10 17:57:52.303  9066  9066 D Foobar  : 	at S0.a.e(SourceFile:30)
09-10 17:57:52.303  9066  9066 D Foobar  : 	at H.a.k(SourceFile:160)
09-10 17:57:52.303  9066  9066 D Foobar  : 	at m.n.m(SourceFile:132)
09-10 17:57:52.303  9066  9066 D Foobar  : 	at Z0.a.d(SourceFile:9)
09-10 17:57:52.303  9066  9066 D Foobar  : 	at o1.v.n(SourceFile:81)
```

Will become:
```
09-10 17:57:52.303  9066  9066 D Foobar  : 	at com.example.myapplication.Foo.foo(Foo.java:7)
09-10 17:57:52.303  9066  9066 D Foobar  : 	at com.example.myapplication.MainActivity.onClick(MainActivity.java:38)
09-10 17:57:52.303  9066  9066 D Foobar  : 	at com.example.myapplication.MainActivity.Greeting$lambda$1$lambda$0(MainActivity.java:43)
09-10 17:57:52.303  9066  9066 D Foobar  : 	at S0.MainActivity$$ExternalSyntheticLambda0.invoke(MainActivity.java:0)

09-10 17:57:52.303  9066  9066 D Foobar  : 	at androidx.compose.foundation.ClickablePointerInputNode$pointerInput$3.invoke-k-4lQ0M(ClickablePointerInputNode.java:987)
09-10 17:57:52.303  9066  9066 D Foobar  : 	at androidx.compose.foundation.ClickablePointerInputNode$pointerInput$3.invoke(ClickablePointerInputNode.java:981)

09-10 17:57:52.303  9066  9066 D Foobar  : 	at androidx.compose.foundation.gestures.TapGestureDetectorKt$detectTapAndPress$2$1.invokeSuspend(TapGestureDetectorKt.java:255)

09-10 17:57:52.303  9066  9066 D Foobar  : 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(BaseContinuationImpl.java:33)

09-10 17:57:52.303  9066  9066 D Foobar  : 	at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTaskKt.java:179)
```

We need to either use an `if (retracedFrames.hasNext())` as in my PR or maybe better, to change line 182 to use `print` instead of `println`:

```
            stackTraceWriter.print(deobf);
```

I'm not sure what you mean about line 25. It seems to work without changing it.

I'm updating the PR.

Reply:
> I'm not sure what you mean about line 25. It seems to work without changing it.

I meant the line here:

```
            // Print out the original line.
            result.append(obfuscatedLine);
```

which used to add a newline before #200 : 

```
                // Print out the original line.
                stackTraceWriter.println(obfuscatedLine);
```

Reply:
With your latest change, if I take the following simple stacktrace and mapping, everything is printed on one line, so your original suggestion of adding the new line only `if (retracedFrames.hasNext())` seems to be correct.

```
Caused by: java.lang.RuntimeException: AAA
	at com.example.demo.sub.TestComponent2.a(Unknown Source)
	at com.example.demo.DemoApplication.main(Unknown Source)

com.example.demo.DemoApplication -> com.example.demo.DemoApplication:
    void <init>() -> <init>
    void main(java.lang.String[]) -> main
com.example.demo.TestComponent -> com.example.demo.TestComponent:
    void <init>() -> <init>
    int getAnswer() -> a
com.example.demo.sub.TestComponent2 -> com.example.demo.sub.TestComponent2:
    void <init>() -> <init>
    java.lang.String getMessage() -> a
```

Reply:
Hmm, for me, it works fine with the latest version.

I'm on a Linux machine. What system are you on?